java/ql/src/Violations of Best Practice/Dead Code/DeadStoreOfLocal.ql

```diff
@@ -27,13 +27,24 @@ predicate flowStep(Expr decl, Expr init) {
   decl.(CastExpr).getExpr() = init
 }
 
+predicate isDefaultValueLiteral(Literal l, Type t) {
```

Comment:
To resolve the cartesian product worry (all reftypes X all nulls), how about just omitting the Type column? This could be just:

Don't flag Expr e if e is (a) null, (b) zero, or (c) of integral type and negative one. Note this actually expands the query slightly to permit `Integer i = 0;`, but considering we already permit `Integer i = -1;` then we should probably accept that anyhow.

Reply:
Sorry for the latest response. Was this a suggestion for me or directed to aschackmull, or both of us?